src/App.js

```diff
@@ -29,32 +66,260 @@ class App extends React.Component {
     });
   }
 
-  // Note use of array fields syntax to avoid having to manually bind this method to the class
-  writeData = () => {
-    const messageListRef = ref(database, DB_MESSAGES_KEY);
-    const newMessageRef = push(messageListRef);
-    set(newMessageRef, "abc");
+  handleChange = (e) => {
+    this.setState({ textInputValue: e.target.value });
+  };
+
+  // writeData = (url) => {
+  //   const messageListRef = ref(database, DB_MESSAGES_KEY);
+  //   const newMessageRef = push(messageListRef);
+  //   set(newMessageRef, {
+  //     message: this.state.textInputValue,
+  //     date: new Date().toLocaleString(),
+  //     url: url,
+  //     numberOfLikes: this.state.numberOfLikes,
+  //   });
+
+  //   this.setState({
+  //     textInputValue: "",
+  //     fileInputFile: null,
+  //     fileInputValue: "",
+  //   });
+  // };
+
+  submit = (event) => {
+    event.preventDefault();
+    const fullStorageRef = storageRef(
+      storage,
+      STORAGE_KEY + this.state.fileInputFile.name
+    );
+
+    uploadBytes(fullStorageRef, this.state.fileInputFile).then(() => {
+      getDownloadURL(fullStorageRef, this.state.fileInputFile.name)
+        .then((url) => {
+          this.writeData(url);
+        })
+        .catch((err) => console.log(err));
+    });
+  };
+
+  deleteMessage = (key) => {
+    const messageRef = ref(database, `${DB_MESSAGES_KEY}/${key}`);
+
+    remove(messageRef)
+      .then(() => {
+        console.log("Data deleted");
```

Comment:
all the console.logs are not useful. I would remove them. If you want error handling and the like, use state and do something meaningful for the user with it